### PR TITLE
Some improvements to the RCAIt

### PR DESCRIPTION
**Pull request by yojs**
*Fixes #:* #368

*Description of changes:*
1. Now we should be able to run a test method from Intellij
2. The validations checks for no ERROR in logs. The test fails if some errors are generated.
3. An ability to run all the integration tests using the target ./gradlew rcaIt
4. Some code cleanups.

*Tests:*
These are changes to test framework. All current tests passing should be good enough.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -183,6 +183,16 @@ task rcaTest(type: Test) {
...
 }
 
+task rcaIt(type: Test) {
+    dependsOn checkstyleMain
+    dependsOn checkstyleTest
+    dependsOn spotbugsMain
+    useJUnit {
+        includeCategories 'com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.integTests.framework.RcaItMarker'
+    }
+    //testLogging.showStandardStreams = true
```
Comment (rguo-aws) [inline build.gradle:193]: remove this line ?

Comment (yojs) [reply build.gradle:193]: Wanted to keep it so than people can enable it if they want stdout ?

File: docs/rcait.md [hunk 2]
```diff
@@ -75,7 +79,57 @@ of 60 seconds.
...
+0 | HostTag.DATA_0
+
+### Examples
+
+Some of the examples for how to write the integ tests can be found here:
+`src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/integTests/tests/poc/*`
+
+### Running the integ tests
+1. All integration tests written using this framework can be run by using the command
+`./gradlew rcaIt`
```
Comment (ktkrg) [inline docs/rcait.md:126]: maybe we can make the command a verb like other gradle tasks? `runRcaIt`?

File: build.gradle [hunk 3]
```diff
@@ -183,6 +183,16 @@ task rcaTest(type: Test) {
     testLogging.showStandardStreams = true
 }
 
+task rcaIt(type: Test) {
```
Comment (ktkrg) [inline build.gradle:186]: Maybe we can make this a verb like other gradle tasks(build, assemble, etc) `runRcaIt`?

**Pull request merged**